operations/companies.md

```diff
@@ -47,152 +50,231 @@ Note this operation uses [Pagination](../guidelines/pagination.md) and supports
 | `AccessToken` | string | required | Access token of the client application. |
 | `Client` | string | required | Name and version of the client application. |
 | `ChainIds` | array of string | optional, max 1000 items | Unique identifiers of the chain. If not specified, the operation returns data for all chains within scope of the Access Token. |
-| `Ids` | array of string | optional, max 1000 items | Unique identifiers of [Companies](#company). |
-| `Names` | array of string | optional, max 1000 items | Names of [Companies](#company). |
-| `CreatedUtc` | [Time interval](_objects.md#time-interval) | optional, max length 3 months | Interval of [Company](#company) creation date and time. |
-| `UpdatedUtc` | [Time interval](_objects.md#time-interval) | optional, max length 3 months | Interval of [Company](#company) last update date and time. |
-| `ExternalIdentifiers` | array of string | optional, max 1000 items | Identifiers of [Company](#company) from external system. |
-| `Limitation` | [Limitation](../guidelines/pagination.md#limitation) | required | Limitation on the quantity of customers returned. |
+| `MotherCompanyIds` | array of string | optional, max 1000 items | Unique identifiers of mother `Company`. |
+| `CreatedUtc` | [Time interval](_objects.md#time-interval) | optional, max length 3 months |  |
+| `UpdatedUtc` | [Time interval](_objects.md#time-interval) | optional, max length 3 months |  |
+| `Ids` | array of string | optional, max 1000 items | Unique identifiers of [Companies](companies.md#company). |
+| `ExternalIdentifiers` | array of string | optional, max 1000 items | Identifiers of [Company](companies.md#company) from external system. |
+| `Name` | string | optional |  |
```

Comment:
❓Question: Whats the difference between `Name` and `Names`?

Reply:
After finding out that there is also `ActivityStates` property and that `Extent` is used as basically a default/fallback value for cases when `ActivityStates` is `null`. I wonder how do ppl know  how to use this endpoint 😅 

Should I try to come up with some description that will explain this madness?

Reply:
fixed by marking `name` as InternalImplementation